### ISSUE TITLE
the source code

### ISSUE THREAD

**Issue by chensming** [longzilicart/pip_universal#1]
Hi, bro, nice jobs! When will you release the source code?

**Comment by longzilicart** [comment]
Hi, I found the paper is online now. I will update the code in a few days.

**Comment by chensming** [comment]
"During training, we utilized cropped patches with a size of 128 × 128 for 150 epochs and 200 × 200 for the rest 50 epochs, using random horizontal and vertical flips as the only augmentation.".
I took the liberty of asking how you implemented this code. Also,  I would then like to know what your batch size is set to for the second stage. Thanks in advance.

**Comment by longzilicart** [comment]
It is implemented by simply resuming the network and optimizer with a larger patch size. The total batch size is set to 16 then. This primarily benefits image deblurring, as most methods typically use a 256 patch size. However, to support datasets with resolutions smaller than 256, I've opted for a 200 patch size, ensuring support for all datasets.

**Comment by chensming** [comment]
> It is implemented by simply resuming the network and optimizer with a larger patch size. The total batch size is set to 16 then. This primarily benefits image deblurring, as most methods typically use a 256 patch size. However, to support datasets with resolutions smaller than 256, I've opted for a 200 patch size, ensuring support for all datasets.

In other words, you save the weights and then train them in resume, right?

**Comment by longzilicart** [comment]
Correct, we only involve the most simple training procedure without any special tricks. It also works well under a fixed patch size.

**Comment by chensming** [comment]
> Correct, we only involve the most simple training procedure without any special tricks. It also works well under a fixed patch size.

Okay, Thanks. 
I'm a little confused about the loss function, the $L_{ddl}$ loss, can you tell me in which section it is mentioned, I read the paper and couldn't find it. 
Then I saw that the ablation study session also had a $\theta_{thre}$, and I didn't understand what it meant either.

![image](https://github.com/longzilicart/pip_universal/assets/45591767/5a641d1d-d65a-46a6-a2af-ba6ee83cc35b)

**Comment by longzilicart** [comment]
It is in eq(4), a constraint loss for the degradation-aware prompt.

**Comment by chensming** [comment]
> It is in eq(4), a constraint loss for the degradation-aware prompt.

Oh, sorry, I saw that. Thank you so much.

**Comment by longzilicart** [comment]
you are welcome

**Comment by xuhangc** [comment]
Any updates?

**Comment by longzilicart** [comment]
> Any updates?

I'm really sorry for the delay. I've been caught up with a bunch of ddl lately. It might take me a few more days to focus on this (in this week). If you've got any questions or suggestions please don't hesitate to email me.
